### PR TITLE
fix reload twice bug

### PR DESCRIPTION
**Pull request by zroehr**
fix reload twice bug
fix page jump after hash change
fix index term cover up by header

# Screencast

Sorry it's slow because it is running in Dev mode and Phil's trackpad is very old

cnx-back-button

### CODE REVIEW

**Comment by philschatz** [comment]
Awesome! I didn't think the 3-sentence description did it justice so I added a screencast to the Pull Request Body above

cnx-back-button

File: src/scripts/modules/media/media.coffee [hunk 1]
```diff
@@ -111,7 +114,7 @@ define (require) ->
       adjustHashTop = ->
         handleHeaderViewPinning()
         if isPinned
-          obscured = $pinnable.height()
+          obscured = $pinnable.height() + 50
```
Comment (philschatz) [inline src/scripts/modules/media/media.coffee]: How well does this works when the window width is smaller (like on a mobile phone)?

If this number will continue to be used can it be moved into a common file so it is not duplicated?

**Comment by zroehr** [comment]
Thanks Phil!

File: src/scripts/modules/media/media.coffee [hunk 2]
```diff
@@ -67,6 +67,11 @@ define (require) ->
 
     triggerHashChange: (e) ->
       Backbone.trigger('window:hashChange')
+      href = $(e.currentTarget).attr('href')
+
+      e.preventDefault()
+      $elHash = href.split("#")[1]
```
Comment (philschatz) [inline src/scripts/modules/media/media.coffee]: Is this variable used anywhere?

File: src/scripts/modules/media/body/body.coffee [hunk 3]
```diff
@@ -445,11 +445,11 @@ define (require) ->
...
       # to force the browser window to find the intended content (as a side effect)
       jumpToHash = () =>
-        if currentPage.get('loaded') and not @fragmentReloaded and window.location.hash
-          @fragmentReloaded = true
-
+        if currentPage.get('loaded') and window.location.hash
           hash = window.location.hash
-          window.location.replace(hash)
+          $pinnable = $('.pinnable')
+          obscured = $pinnable.height() + 50
```
Comment (philschatz) [inline src/scripts/modules/media/body/body.coffee]: How well does this works when the window width is smaller (like on a mobile phone)?

If this number will continue to be used can it be moved into a common file so it is not duplicated?

**Comment by caderitter** [comment]
So happy this got fixed 😄  thanks so much for jumping on this!!

File: src/scripts/modules/media/media.coffee (hunk 2 above)
Comment (zroehr) [inline src/scripts/modules/media/media.coffee]: good catch, this was left over from some code I removed and is not used.

File: src/scripts/modules/media/body/body.coffee (hunk 3 above)
Comment (zroehr) [inline src/scripts/modules/media/body/body.coffee]: I will verify this works on small screens and move it into a common file

File: src/scripts/modules/media/media.coffee (hunk 1 above)
Comment (zroehr) [inline src/scripts/modules/media/media.coffee]: I will verify this works on small screens and move it into a common file

**Comment by philschatz** [comment]
The link scroll seems to not move the target item to the top of the page. See:

cnx-link-scroll

Is that the desired behavior?

**Comment by zroehr** [comment]
I added some offset so the link wouldn't sit right at the bottom of the header after a click. This offset also allows for feature boxes to be shown in full when a jump link is linking to a note feature. I can adjust this if needed.

**Comment by zroehr** [comment]
@philschatz heres an example of what im referring to when it comes to notes

**Comment by kerwinso** [comment]
Dev testing failed. See https://github.com/Connexions/cnx-recipes/issues/213 and https://github.com/Connexions/webview/issues/1579. Also, a minor regression found in https://github.com/Connexions/webview/issues/1578.

**Comment by kerwinso** [comment]
Passed QA:

- Hash links go where they're supposed to (including footnote references) without reloading the page. 
- Back button in the browser works, even when navigating between 2 hash links. 
- Index terms and other content arrived at via hash link are no longer hidden by the top nav header. 

Such a better user experience!

**Pull request merged**